File: HiddenDigits/hd.rb
# hidden digits soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line.chomp!
    result = ""
    line.split("").each do |char|
        ord = char.ord
        if (ord >= 48 && ord <= 57)
            result << (ord - 48).to_s
        elsif (ord >= 97 && ord <= 106)
            result << (ord - 97).to_s
        end
    end
    if result.empty?
        puts "NONE"
    else
        puts result
    end
end

File: HiddenDigits/py3/hd.py
# hidden digits soln in py3 for codeeval by steven a dunn

import sys

f = open(sys.argv[1], 'r')
for line in f:
    result = ""
    line = list(line)
    for char in line:
        ordVal = ord(char)
        if (ordVal >= 48 and ordVal <= 57):
            result += str(ordVal - 48)
        elif (ordVal >= 97 and ordVal <= 106):
            result += str(ordVal - 97)
    if not result:
        print ("NONE")
    else:
        print (result)
f.close()